src/index.ts

```diff
@@ -73,10 +150,90 @@ class AsyncSource<T> {
         this.isFetchedData = false;
         this.responseData = null;
         this.lastRequestId = 0;
+
+        const cacheKey = this.getCacheKey();
+
+        if (cacheKey) {
+            new Promise((resolve) => resolve(this.storage?.removeItem(cacheKey)));
+        }
+    }
+
+    private async loadFromStorage() {
+        const cachedData = await this.getCachedData();
+        if (cachedData) {
+            this.responseData = cachedData;
+            this.isFetchedData = true;
+            this.isRequestPending = false;
+        }
+    }
+
+    private getCacheKey() {
+        if (!this.cacheKey) return;
+
+        if (this.args) {
+            return `${this.cacheKey}-${this.args}`;
+        }
+        return this.cacheKey;
+    }
+
+    private async getCachedData(): Promise<PromiseResult<T> | null> {
+        const cacheKey = this.getCacheKey();
+
+        if (!cacheKey) return null;
+
+        const storedData = await new Promise<string | null | undefined>((resolve) => resolve(this.storage?.getItem(cacheKey)));
+
+        if (storedData) {
+            try {
+                const { data, timestamp } = JSON.parse(storedData);
+                const isExpired = Date.now() - timestamp > this.cacheTime;
+                return isExpired ? null : (data as PromiseResult<T>);
+            } catch (error) {
+                this.onError(error as Error);
+            }
+        }
+        return null;
+    }
+
+    private setArgs(args: Array<any>) {
+        if (args) {
+            try {
+                this.args = JSON.stringify(args);
+            } catch (error) {
+                this.onError(error as Error);
+            }
+        }
     }
 
-    // Core request method
-    private async request(args: Array<any>, successHandler?: ((response: T) => void), isImmediate?: boolean) {
+    private setCachedData(data: T): void {
+        const cacheKey = this.getCacheKey();
+
+        if (!cacheKey) return;
+
+        const cacheValue = {
+            data,
+            timestamp: Date.now()
+        };
+
+        try {
+            new Promise((resolve) => resolve(this.storage?.setItem(cacheKey, JSON.stringify(cacheValue))));
+        } catch (error) {
+            this.onError(error as Error);
+        }
+    }
+
+    private async request(args: Array<any>, successHandler?: (response: T) => void, isImmediate?: boolean): Promise<void> {
+        this.setArgs(args);
```

Comment:
1. Add additional check for enabled state to skip current cache logic when cache is not configured
2. Move current after creating requestId